res/logback.xml

```diff
@@ -0,0 +1,26 @@
+<configuration>
+    <appender name="FILE" class="ch.qos.logback.core.rolling.RollingFileAppender">
+        <file>briefcase.log</file>
+        <rollingPolicy class="ch.qos.logback.core.rolling.SizeAndTimeBasedRollingPolicy">
+            <fileNamePattern>briefcase.%d{yyyy-MM-dd}.%i.log</fileNamePattern>
+            <!-- keep 14 days' worth of history capped at 1GB total size -->
+            <maxHistory>10</maxHistory>
+            <maxFileSize>10MB</maxFileSize>
+            <totalSizeCap>100MB</totalSizeCap>
+        </rollingPolicy>
+        <encoder>
+            <pattern>[%date] [%level] [%thread] [%logger{10}] %msg%n</pattern>
+        </encoder>
+    </appender>
+
+    <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
+        <encoder>
+            <pattern>[%date] [%level] [%thread] [%logger{10}] %msg%n</pattern>
+        </encoder>
+    </appender>
+
+    <root level="debug">
```

Comment:
Are we sure we want the default to be debug? That's pretty chatty! Maybe warn for the console and debug for the log file?

Reply:
I changed it to info for the console since I think it's good to see some informational message in the console, no? :)

Reply:
This discussion highlights why I said that getting clever with the logging config was the wrong solution. The reality is that when using CLI-mode you would want different console output than when using it in GUI-mode. You can accomplish this in a few ways, like programmatically manipulating the logging config at runtime. However, IMO it is better to simplify the design so it actually makes sense; use the console output streams System.out and err to provide CLI-specific messages. Logging should be used for other diagnostics, and sparingly (at least in terms of I/O - logging frameworks like this exist to make it possible to add good logging code that can later be turned on without causing performance issues... until the deeper diagnostics are necessary).

By trying to reduce this to a single problem, you conflate the meaning of the configs. You are going to have to carefully manage them in order to prevent inadvertently squelching and/or polluting the CLI output by removing/adding logging code (or changing levels) in other classes. You're increasing the complexity and cognitive load for a low-quality utility. That is simply not good design.

Ideally, in all logging modes you limit logging output to only relevant information. In practice, determining relevance can be difficult due to the fact you don't know perfectly in advance what the problems will be. However, to actually output routine information just adds unnecessary I/O that will slow the application down (which can be severe). You can (and should) log that type of information at debug or trace levels and squelch them by default. Logging routine information like this also makes diagnosing issues more difficult because you have to sift through massive volumes of "normal" output to find interesting information. This is true even if you can squelch them, since when you need to enable debug (or other levels) logging, you're still going to want quality messages. The moral is to carefully design the logging mode as a user experience in itself. Here we're doing the opposite - merging multiple experiences creating a single (lousy) one. The Apache HTTP Components logging is a great example of logging used well. It was clear they were purposeful in their use of logging, briefcase's use is not (yet).

As much as possible, we should instead try to provide direct feedback to users (through console output on CLI and graphical controls in GUI) rather than forcing users to sift through log output to figure out what's going on (and devs to keep both modes operating well). You will have full control over the user experience by not merging CLI and GUI output. You also can take full advantage of the strengths of the respective application modes and intent is clear from the code.

Reply:
I'm not sure I catch everything here.

For me personally, I like to see some informational messages getting printed in my console when I run Java application. To me seeing those messages imply progress, there's something that the system is doing currently.

For CLI, I agree that the reply the user is getting should be in the System.out or System.err instead of getting log messages printed in front of them. Isn't that what #209 should be addressing? And I still see the benefit of CLI application create log file with more information that what the user is getting through System.out and System.err.

Reply:
@nribeka My overall feedback is that we should consider the resource's purpose to the user, in this case the console, and make usage the most effective considering the purpose. I don't think this is considerate of purpose. The console's utility to CLI-mode is different than in GUI-mode when launched from the command line. It changes again when launched from click-mode, where the console is entirely hidden.

I understand that a moving console can be comforting for many devs while developing. I am arguing your feeling of comfort may be misplaced: it's simply not effective use of the medium. An info message or two at startup is fine, but ideally you want feedback to be presented where the user will expect it and can use it: in the primary user interface. The current interface reduces the value of the console by raising the noise threshold past the point where normal users will venture to comprehend it. Add to consideration the performance degradation (for quite possibly invisible, messages) logging at inappropriate log levels and it enters the realm of "this project doesn't know what it is doing". We also would double-log these, once in the console and again in a persistent log. That is uninspiring and honestly, silly. 

A better alternative is showing only a single message at ERROR with the form uuid and plain-language message (user-actionable ideally) rather than 1000s of status messages with one error message somewhere in the noise.

This shouldn't be contentious. The reason for the framework is to allow for power-users or devs to override the defaults, but the default should be intelligent for common users.

* Fix the log levels to be more appropriate. For example, INFO shouldn't be active unless it's actually informative. Logging the form fetch for 100s or 1000s of forms isn't informative.
* Fix the default logging config to use the console sparingly (perhaps at WARN, or INFO once the logging level usages are fixed)
* Decouple CLI-mode output from the logging. Whether or not you want to log at INFO or not should not remove the primary output from CLI-mode. It is a misuse of the tool.